src/aktualizr_secondary/aktualizr_secondary.cc

```diff
@@ -1,73 +1,75 @@
+#include "uptane/manifest.h"
 #include "utilities/utils.h"
 
-class SecondaryAdapter : public Uptane::SecondaryInterface {
- public:
-  SecondaryAdapter(AktualizrSecondary& sec) : secondary(sec) {}
-  ~SecondaryAdapter() override = default;
-
-  Uptane::EcuSerial getSerial() override { return secondary.getSerialResp(); }
-  Uptane::HardwareIdentifier getHwId() override { return secondary.getHwIdResp(); }
-  PublicKey getPublicKey() override { return secondary.getPublicKeyResp(); }
-  Json::Value getManifest() override { return secondary.getManifestResp(); }
-  bool putMetadata(const Uptane::RawMetaPack& meta_pack) override {
-    return secondary.putMetadataResp(Metadata(meta_pack));
-  }
-  int32_t getRootVersion(bool director) override { return secondary.getRootVersionResp(director); }
-  bool putRoot(const std::string& root, bool director) override { return secondary.putRootResp(root, director); }
-  bool sendFirmware(const std::shared_ptr<std::string>& data) override {
-    return secondary.AktualizrSecondary::sendFirmwareResp(data);
-  }
+#include <sys/types.h>
+#include <memory>
 
- private:
-  AktualizrSecondary& secondary;
-};
-
-AktualizrSecondary::AktualizrSecondary(const AktualizrSecondaryConfig& config,
-                                       const std::shared_ptr<INvStorage>& storage)
-    : AktualizrSecondaryCommon(config, storage),
-      socket_server_(std_::make_unique<SecondaryAdapter>(*this), SocketFromPort(config.network.port)) {
-  // note: we don't use TlsConfig here and supply the default to
-  // KeyManagerConf. Maybe we should figure a cleaner way to do that
-  // (split KeyManager?)
+AktualizrSecondary::AktualizrSecondary(AktualizrSecondaryConfig config, std::shared_ptr<INvStorage> storage,
+                                       std::shared_ptr<KeyManager> key_mngr, std::shared_ptr<UpdateAgent> update_agent)
+    : config_(std::move(config)),
+      storage_(std::move(storage)),
+      keys_(std::move(key_mngr)),
+      update_agent_(std::move(update_agent)) {
   if (!uptaneInitialize()) {
     LOG_ERROR << "Failed to initialize";
     return;
   }
-}
 
-void AktualizrSecondary::run() {
-  connectToPrimary();
-  socket_server_.Run();
-}
-
-void AktualizrSecondary::stop() { /* TODO? */
+  manifest_issuer_ = std::make_shared<Uptane::ManifestIssuer>(keys_, ecu_serial_);
+
+  if (hasPendingUpdate()) {
+    // TODO: refactor this to make it simpler as we don't need to persist/store
+    // an installation status of each ECU but store it just for a given secondary ECU
+    std::vector<Uptane::Target> installed_versions;
+    boost::optional<Uptane::Target> pending_target;
+    storage_->loadInstalledVersions(ecu_serial_.ToString(), nullptr, &pending_target);
+    data::InstallationResult install_res =
+        data::InstallationResult(data::ResultCode::Numeric::kUnknown, "Unknown installation error");
+    LOG_INFO << "There is a pending update, try to apply it, update hash: " << pending_target_.sha256Hash();
```

Comment:
Might be worth making this just a bit clearer. "Try to apply it" sounds like the user should take action. Maybe rephrase to "Pending update found; attempting to apply it. Target hash: ".

Reply:
ok